Modules/itertoolsmodule.c

```diff
@@ -4767,30 +4675,18 @@ itertoolsmodule_exec(PyObject *mod)
     ADD_TYPE(mod, state->filterfalse_type, &filterfalse_spec);
     ADD_TYPE(mod, state->groupby_type, &groupby_spec);
     ADD_TYPE(mod, state->_grouper_type, &_grouper_spec);
+    ADD_TYPE(mod, state->islice_type, &islice_spec);
     ADD_TYPE(mod, state->pairwise_type, &pairwise_spec);
     ADD_TYPE(mod, state->permutations_type, &permutations_spec);
     ADD_TYPE(mod, state->product_type, &product_spec);
+    ADD_TYPE(mod, state->repeat_type, &repeat_spec);
     ADD_TYPE(mod, state->starmap_type, &starmap_spec);
     ADD_TYPE(mod, state->takewhile_type, &takewhile_spec);
+    ADD_TYPE(mod, state->tee_type, &tee_spec);
+    ADD_TYPE(mod, state->teedataobject_type, &teedataobject_spec);
     ADD_TYPE(mod, state->ziplongest_type, &ziplongest_spec);
 
-    PyTypeObject *typelist[] = {
-        &batched_type,
-        &islice_type,
-        &chain_type,
-        &repeat_type,
-        &tee_type,
-        &teedataobject_type
-    };
-
-    Py_SET_TYPE(&teedataobject_type, &PyType_Type);
-
-    for (size_t i = 0; i < Py_ARRAY_LENGTH(typelist); i++) {
-        if (PyModule_AddType(mod, typelist[i]) < 0) {
-            return -1;
-        }
-    }
-
+    Py_SET_TYPE(state->teedataobject_type, &PyType_Type);
```

Comment:
Not entirely sure why this is needed.

Reply:
Me neither, but I kept it in place anyway.

Reply:
Added in f3938fd02902ca55d471f9fbb533e84560745e6e. Not needed anymore. We can purge it in a follow-up PR.